src/cli/telemetry/telemetry.ts

```diff
@@ -0,0 +1,304 @@
+    typescript: versions.typescript,
+    rollup: versions.rollup,
+  };
+
+  await sendMetric('stencil_cli_command', data);
+  logger.debug(`${logger.blue('Telemetry')}: ${logger.gray(JSON.stringify(data))}`);
+
+  if (error) {
+    throw error;
+  }
+}
+
+function isUsingYarn() {
+  return getCompilerSystem().getEnvironmentVar('npm_execpath').includes('yarn');
+}
+
+async function getActiveTargets(config: Config): Promise<string[][] | OutputTargetOptions[][]> {
+  return (config.outputTargets as any[]).map(target => {
+    let outputTargetSettings: OutputTargetOptions = {};
+
+    if (target.type === 'www') {
+      outputTargetSettings.serviceWorker = isObject(target?.serviceWorker);
+    }
+
+    return [target.type, outputTargetSettings];
+  });
+}
+
+/**
+ * Reads package-lock.json and package.json files in order to cross references the dependencies and devDependencies properties. Pull the current installed version of each package under the @stencil, @ionic, and @capacitor scopes.
+ * @returns string[]
+ */
+async function getInstalledPackages() {
+  try {
+    // Read package.json and package-lock.json
+    const appRootDir = getCompilerSystem().getCurrentDirectory();
+
+    const packageJson: PackageJsonData = await tryFn(
+      readJson,
+      getCompilerSystem().resolvePath(appRootDir + '/package.json'),
+    );
+
+    const packageLockJson: any = await tryFn(
+      readJson,
+      getCompilerSystem().resolvePath(appRootDir + '/package-lock.json'),
+    );
+
+    // They don't have a package.json for some reason? Eject button.
+    if (!!packageJson) {
+      return [];
+    }
+
+    const packages = Object.entries({
+      ...packageJson.devDependencies,
+      ...packageJson.dependencies,
+    }) as [string, string][];
+
+    // Collect packages only in the stencil, ionic, or capacitor org's:
+    // https://www.npmjs.com/org/stencil
+    const ionicPackages = packages.filter(
+      ([k]) => k.startsWith('@stencil/') || k.startsWith('@ionic/') || k.startsWith('@capacitor/'),
+    );
+
+    const versions = packageLockJson
+      ? ionicPackages.map(
+          ([k, v]) =>
+            `${k}@${
+              (packageLockJson?.dependencies[k] as { version: string })?.version ??
+              (packageLockJson?.devDependencies[k] as { version: string })?.version ??
+              v
+            }`,
+        )
+      : ionicPackages.map(([k, v]) => `${k}@${v}`);
+
+    return versions;
+  } catch (err) {
+    hasDebug() && console.error(err);
+    return [];
+  }
+}
+
+/**
+ * If telemetry is enabled, send a metric via IPC to a forked process for uploading.
+ */
+export async function sendMetric(name: string, value: TrackableData): Promise<void> {
+  const message: Metric = {
+    name,
+    timestamp: new Date().toISOString(),
+    source: 'stencil_cli',
+    value,
+  };
+
+  await sendTelemetry({ type: 'telemetry', message });
+}
+
+/**
+ * Used to read the ~/.ionic/config.json file's tokens.telemetry property.
```

Comment:
Food for thought - the read of ` ~/.ionic/config.json`, specifically the file name, is abstracted out to `readConfig`. Should the name/location of that file change, we may not remember to update this comment (which would make it stale/misinformation)